Mage/src/main/java/mage/util/CardUtil.java

```diff
@@ -1625,4 +1625,35 @@ public static String convertStartingLoyalty(int startingLoyalty) {
                 return "" + startingLoyalty;
         }
     }
+
+    public static String getAddRemoveCountersLogMessage(Game game, Ability source, UUID playerId, int numCounters, String counterName, String targetName, boolean remove) {
+        StringBuilder sb = new StringBuilder();
+        MageObject sourceObject = game.getObject(source);
+        if (sourceObject != null) {
+            sb.append(sourceObject.getLogName());
+            sb.append(": ");
+        }
+        Player addingPlayer = game.getPlayer(playerId);
+        if (addingPlayer != null) {
```

Comment:
Are there instances where `addingPlayer == null` is valid?

Reply:
Also, I think you always have to handle `game.getPlayer()` potentially returning null.

Reply:
Let me rephrase: is getting `addingPlayer == null` a bug? Is it supposed to happen?

If it's not supposed to happen, we should error out rather than log the message.